lib/features/user/data/datasources/user_remote_data_source.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:coffeecard/core/errors/failures.dart';
+import 'package:coffeecard/core/network/network_request_executor.dart';
+import 'package:coffeecard/features/user/data/models/user_model.dart';
+import 'package:coffeecard/generated/api/coffeecard_api.swagger.dart';
+import 'package:coffeecard/generated/api/coffeecard_api_v2.swagger.dart'
+    hide MessageResponseDto;
+import 'package:coffeecard/models/account/update_user.dart';
+import 'package:dartz/dartz.dart';
+
+class UserRemoteDataSource {
+  final CoffeecardApi apiV1;
+  final CoffeecardApiV2 apiV2;
+  final NetworkRequestExecutor executor;
+
+  UserRemoteDataSource({
+    required this.apiV1,
+    required this.apiV2,
+    required this.executor,
+  });
+
+  /// Get the currently logged in user.
+  Future<Either<NetworkFailure, UserModel>> getUser() async {
+    final result = await executor(() => apiV2.apiV2AccountGet());
+
+    return result.map(UserModel.fromResponse);
+  }
+
+  /// Updates the details of the currently logged in user based on
+  /// the non-null details in [user]
+  Future<Either<NetworkFailure, UserModel>> updateUserDetails(
+    UpdateUser user,
+  ) async {
+    final result = await executor(
+      () => apiV2.apiV2AccountPut(
+        body: UpdateUserRequest(
+          name: user.name,
+          programmeId: user.occupationId,
```

Comment:
unrelated to this particular PR but perhaps we should change the api to accept `occuptionId` instead of `programmeId`